rfc/rfc-81/rfc-81.md

```diff
@@ -0,0 +1,108 @@
+LogCompaction with Merge Sort is introduced to achieve lightweight minor compaction by merging records through N-way streaming of ordered data, thus improving the writing performance of the write side. At the same time, thanks to the ordered data, the query performance on the read side can be improved when the primary key is met.
+
+## Implementation
+### HoodieConfig etc
+Added a new configuration item for the HoodieLogBlock streaming read buffer size. The default value is 10MB.
+#### Flink
+Added a new configuration item to enable LogCompaction. The default value is false.
+Note:
+1. After Flink enables LogCompaction, the default implementation is MergeSort.
+2. Currently, the log format only supports AVRO. After enabling LogCompaction, need to check whether the hoodie.logfile.data.block.format configuration item is correct.
+
+#### Spark
+Added a new configuration item for whether to enable MergeSort in LogCompaction. The default value is false.
+Note:
+1. Currently, the log format only supports AVRO. After LogCompaction turns on MergeSort, you need to check whether the hoodie.logfile.data.block.format configuration item is correct.
+
+### DeltaCommit
+#### Flink
+When LogCompaction is enabled, DeltaCommit sorts the written data by RecordKey to achieve orderly records in LogBlock.
+#### Spark
+When LogCompaction turns on MergeSort, DeltaCommit sorts the written data by RecordKey to achieve orderly records in LogBlock.
+
+### LogCompaction
+The following contents are all about enabling LogCompaction or enabling MergeSort with LogCompaction.
+#### Flink
+Flink has not yet fully implemented the LogCompaction feature, so the operator needs to be implemented:
+1. LogCompactionPlanOperator
+2. LogCompactionOperator
+3. LogCompactionCommitSink
+
+When scanning log files, use the new log scanner: HoodieMergeSortLogRecordScanner to achieve N-way streaming record merging.
+
+#### Spark
+When scanning log files, use the new log scanner: HoodieMergeSortLogRecordScanner to achieve N-way streaming record merging.
+
+### HoodieMergeSortLogRecordScanner
+Implement a min-heap. The heap node is a HoodieLogBlock object. The heap node comparison uses HoodieRecord#RecrodKey.
+When traversing the scanner record iterator, return to the top node of the heap and call the HoodieLogBlock object to return the record.
+
+### HoodieDataBlock
+Added a new streaming read abstract method to help HoodieMergeSortLogRecordScanner avoid OOM when reading N LogBlocks.
+1. HoodieAvroDataBlock: Implements the streaming read member function. Internally, it gradually reads small parts of the log file into the buffer to implement streaming decoding of AVRO records. When the buffer is insufficient, it triggers the reading of the log file. 
+The buffer size can be adjusted by the configuration item.
+2. HFile/Parquet: To be implemented in the future.
+
+## Rollout/Adoption Plan
```

Comment:
It is great if we put some local benchmark numbers in the doc.

Reply:
> It is great if we put some local benchmark numbers in the doc.

Got it, i will add benchmark test